### PR TITLE
[overview] Update to match changes in upstream proposals

### PR DESCRIPTION
**Pull request by rossberg**

### CODE REVIEW

**Comment by Horcrux7** [comment]
The link for Constructed Types is broken.

File: proposals/gc/MVP.md [hunk 1]
```diff
@@ -9,9 +9,9 @@ See [overview](Overview.md) for background.
 
 Based on the following proposals:
 
-* [reference types](https://github.com/WebAssembly/reference-types), which introduces type `anyref` etc.
+* [reference types](https://github.com/WebAssembly/reference-types), which introduces references types
```
Comment (taralx) [inline proposals/gc/MVP.md:12]: typo: references -> reference

File: proposals/gc/MVP.md [hunk 2]
```diff
@@ -20,19 +20,47 @@ All three proposals are prerequisites.
...
 
-* `eqref` is a new reference type
-  - `reftype ::= ... | eqref`
+[Constructed types](https://github.com/WebAssembly/reference-types/blob/master/proposals/function-references/Overview.md) classify the target of a reference and are extended:
 
-* `i31ref` is a new reference type
-  - `reftype ::= ... | i31ref`
+* `any` is a new constructed type
+  - `constype ::= ... | any`
+  - the common supertype of all referencable types
```
Comment (taralx) [inline proposals/gc/MVP.md:29]: The more common spelling of this is 'referenceable'.

**Comment by rossberg** [comment]
Thanks @Horcrux7, @taralx, I pushed a fix.

**Pull request merged**